src/hooks/deleteValidation/useDeleteValidation.js

```diff
@@ -24,12 +24,20 @@ function useDeleteValidation(queryFunction) {
     );
   };
 
-  const deleteValidation = (id) =>
-    products.map((item) =>
-      item.ids.map(
-        (_id) => id === _id && { itemName: item.itemName, itemId: item._id }
-      )
-    );
+  const deleteValidation = (id) => {
+    const validationData = [];
+    for (const item of products) {
+      for (const _id of item.ids) {
+        if (_id === id) {
```

Comment:
I would rather recommend using destruction + map.map.filter or map.reduce. It's crucial to master them. 
  products
	.map( { ids }  => ids
		.filter( _id => _id === id)
		.map(item => {itemID ..., itemname ... }) 
	)
It's also possible to change filter +map with reduce, but readability would not be so great.

Reply:
products.filter((product) => product.ids.includes(id)).map({_id} => _id)

Reply:
products.reduce((acc, item) => { 
if(product.ids.includes(id)) {
acc.push(item._id)
}
returen acc
}, [])